### PR TITLE
Merge tools in grid and layout plots

### PR DESCRIPTION
**Pull request by philippjfr**
In bokeh3 the tool merging is a lot more flexible but by default it is also a lot more conservative. The basic idea behind the new approach is:

> If the properties of the tools match then merge them otherwise don't.

There is a list of properties that are excluded from this logic which covers the 'renderer' and 'overlay' properties by default. Looking at a specific example we see this:

Screen_Shot_2023-03-06_at_11 34 41

- `BoxZoomTool` does not merge because the geo plot has set `match_aspect`
- `WheelZoomTool` does not merge because the geo plot has disabled  `zoom_on_axis`  (because the axes aren't rendered)
- `HoverTool` does not merge because the `tooltips` don't match
- The box select and lasso select tools don't merge because they have different `js_property_callbacks`

With this PR applied we avoid also skip `match_aspect`, `js_property_callbacks` and `zoom_axis` and I'm also considering adding `tooltips` to the list:

Screen Shot 2023-03-06 at 11 59 54

### CODE REVIEW

**Comment by philippjfr** [comment]
After some internal discussion I've decided the logic that makes most sense for us is to always merge tools of the same type  **unless** they define what I've called `disambiguation_properties`, which by default include `name`, `icon`, `tags`, and `description`.

File: holoviews/plotting/bokeh/plot.py [hunk 1]
```diff
@@ -582,11 +582,13 @@ def initialize_plot(self, ranges=None, plots=[]):
             else:
                 passed_plots.append(None)
 
-        toolbar_location = None if bokeh3 else self.toolbar
```
Comment (philippjfr) [inline holoviews/plotting/bokeh/plot.py:585]: @Hoxbro Do you remember why this was needed?

Comment (Hoxbro) [reply holoviews/plotting/bokeh/plot.py:585]: See [this](https://github.com/holoviz/holoviews/pull/5388#issuecomment-1305969977) comment and the next.

Comment (philippjfr) [reply holoviews/plotting/bokeh/plot.py:585]: Fixed the logic in https://github.com/holoviz/holoviews/pull/5640/commits/4a0b16ee21c615d831c04c6044af70c0137a2496

**Comment by Hoxbro** [comment]
61f7c6364ad6ed5b40cccd959b26f6e2afad55bd is to remove toolbars for nested gridplots:

``` python
import holoviews as hv
hv.extension("bokeh")

data = {(i, j): hv.Curve(range(i + j)) for i in range(1, 3) for j in range(2, 4)}
hv.GridSpace(hv.GridSpace(data) + hv.GridSpace(data))
```

| Before | After |
| ---------- | --------|
| image | Screenshot 2023-03-08 12 35 46 |

**Comment by philippjfr** [comment]
Might not be completely done but this is a huge improvement.

**Pull request merged**